sessions/forecasting-nowcasting.qmd

```diff
@@ -0,0 +1,798 @@
+  ) +
+  theme(legend.position = "bottom")
+```
+
+## Quantitative evaluation using scoringutils
+
+Using `scoringutils`, We first create a forecast sample object
+
+```{r forecast-evaluation}
+sc_forecasts <- forecast_results |>
+    left_join(evaluation_data, by = c("actual_day" = "day")) |>
+    group_by(approach, actual_day) |>
+    mutate(
+      .draw = row_number()
+    ) |>
+    as_forecast_sample(
+      forecast_unit = c("approach", "actual_day"),
+      observed = "observed",
+      predicted = ".value",
+      sample_id = ".draw"
+    )
+sc_forecasts
+```
+
+and then calculate the scores:
+
+```{r forecast-evaluation-scores}
+scores <- sc_forecasts |>
+  score()
+
+head(scores)
+```
+
+## Overall performance table
+
+We summarise the performance across all forecast horizons:
+
+```{r performance-table}
+# Create summary table showing overall performance
+overall_performance <- scores |>
+  summarise_scores(by = "approach") |>
+  arrange(crps)
+
+knitr::kable(overall_performance, 
+             caption = "Overall forecast performance by approach",
+             digits = 3)
+```
+
+## Performance by forecast horizon
+
+We examine how performance varies with forecast horizon:
+
+```{r performance-by-horizon}
+# Calculate performance by horizon
+horizon_performance <- scores |>
+  summarise_scores(by = c("approach", "actual_day"))
+
+# Plot showing how performance varies by forecast horizon
+ggplot(horizon_performance) +
+  aes(x = actual_day, y = crps, color = approach) +
+  geom_line(linewidth = 1) +
+  geom_point(size = 1) +
+  scale_y_log10() +
+  labs(
+    x = "Forecast horizon (days)",
+    y = "CRPS (lower is better)",
+    title = "Forecast performance by horizon",
+    color = "Approach"
+  ) +
+  theme_minimal() +
+  theme(legend.position = "bottom")
+```
+
+::: {.callout-tip}
+## Take 5 minutes
+
+Compare the four approaches across the evaluation metrics
+
+:::
+
+::: {.callout-note collapse="true"}
+## Solution
+From this evaluation, we can see:
+
+- **Complete data approach**: Performs much worse overall, with CRPS of 182 compared to ~19-20 for the other approaches. Performance degrades rapidly with longer forecast horizons.
+- **Pipeline approach**: Shows the best overall performance (CRPS = 18.8) but has some systematic underprediction bias (-0.67).
+- **Joint approach**: Slightly worse overall performance (CRPS = 20.5) with some overprediction bias (0.69) and higher dispersion, indicating greater uncertainty.
+
+The pipeline approach performs better here potentially because the reporting delay is relatively short and the central nowcast estimate was accurate.
+In this case, having less uncertainty was beneficial.
+However, this is not guaranteed - the higher dispersion in the joint approach reflects more realistic uncertainty quantification.
+If the nowcast had been less accurate or reporting delays longer, the joint approach might have performed better.
+
+:::
+
+# Going further
```

Comment:
Somewhere, I'm not sure where, I think it is important to mention the limitation of how these methods as we are presenting them only help when there is systematic under-reporting, correct? Or will they extend to settings where there is adjustments down as well, as happens often with ILI?

Either way, I think this distinction is important to make, given the focus on ILI in the new forecasting content, and talking about how these methods can/should/would need to be adjusted to adapt to that setting will help make the link between the two sets of content be clearer.

Reply:
> Or will they extend to settings where there is adjustments down as well, as happens often with ILI?

Yes they can be adjusted. We are only dealing with delayed (note not underreporting) as that is the most common bias we observe in real-time linelists (i.e the focus of the first half of the course).

I think somewhere having a map of how a linelist relates to ILI would be useful as this also guides the modelling choices to make.

I don't really want to go into this level of detail here as I think there is enough content to be going on with. Potentially this makes a good talk or call out.

Reply:
#79